### ISSUE TITLE
About train process and validate result

### ISSUE THREAD

**Issue by JHCheng111** [vinvino02/GLPDepth#13]
Thanks for your impressive work! I've been trying to replicate your work, but there were a strange validate result. Why the metrics suddenly became abnormal and constant from the 11th epoch?
Logs are as follows:
gpu_or_cpu:gpu,	
data_path:../dataset/,	
dataset:nyudepthv2,	
exp_name:test,	
batch_size:12,	
workers:1,	
max_depth:10.0,	
max_depth_eval:10.0,	
min_depth_eval:0.001,	
do_kb_crop:1,	
kitti_crop:None,	
epochs:25,	
lr:0.0001,	
crop_h:448,	
crop_w:576,	
log_dir:./logs,	
val_freq:1,	
save_freq:10,	
save_model:False,	
save_result:False,	

Epoch: 001 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.8535     0.9817     0.9966     0.1276     0.0726     0.4282     0.1548     0.0534     0.1418 
====================================================================================================

Epoch: 002 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.8559     0.9824     0.9962     0.1301     0.0729     0.4214     0.1528     0.0531     0.1376 
====================================================================================================

Epoch: 003 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.8856     0.9853     0.9972     0.1135     0.0619     0.3891     0.1392     0.0474     0.1268 
====================================================================================================

Epoch: 004 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.8847     0.9834     0.9970     0.1113     0.0608     0.3920     0.1403     0.0480     0.1282 
====================================================================================================

Epoch: 005 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.8932     0.9869     0.9976     0.1072     0.0565     0.3765     0.1338     0.0456     0.1219 
====================================================================================================

Epoch: 006 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.9033     0.9871     0.9969     0.1025     0.0534     0.3661     0.1297     0.0440     0.1187 
====================================================================================================

Epoch: 007 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.9005     0.9865     0.9975     0.1050     0.0541     0.3617     0.1303     0.0444     0.1184 
====================================================================================================

Epoch: 008 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.9038     0.9868     0.9974     0.1030     0.0535     0.3565     0.1285     0.0436     0.1169 
====================================================================================================

Epoch: 009 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.9055     0.9879     0.9976     0.1032     0.0521     0.3543     0.1276     0.0435     0.1157 
====================================================================================================

Epoch: 010 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.9060     0.9870     0.9975     0.1020     0.0559     0.3672     0.1299     0.0442     0.1180 
====================================================================================================

Epoch: 011 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0000     0.0000     0.0000     0.9995     2.6884     2.8657     7.8043     3.3855     5.5246 
====================================================================================================

Epoch: 012 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0000     0.0000     0.0000     0.9995     2.6884     2.8657     7.8043     3.3855     5.5246 
====================================================================================================

Epoch: 013 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0000     0.0000     0.0000     0.9995     2.6884     2.8657     7.8043     3.3855     5.5246 
====================================================================================================

Epoch: 014 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0000     0.0000     0.0000     0.9995     2.6884     2.8657     7.8043     3.3855     5.5246 
====================================================================================================

Epoch: 015 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0000     0.0000     0.0000     0.9995     2.6884     2.8657     7.8043     3.3855     5.5246 
====================================================================================================

Epoch: 016 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0000     0.0000     0.0000     0.9995     2.6884     2.8657     7.8043     3.3855     5.5246 
====================================================================================================

Epoch: 017 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0000     0.0000     0.0000     0.9995     2.6884     2.8657     7.8043     3.3855     5.5246 
====================================================================================================

Epoch: 018 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0000     0.0000     0.0000     0.9995     2.6884     2.8657     7.8043     3.3855     5.5246 
====================================================================================================

Epoch: 019 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0000     0.0000     0.0000     0.9995     2.6884     2.8657     7.8043     3.3855     5.5246 
====================================================================================================

Epoch: 020 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0000     0.0000     0.0000     0.9995     2.6884     2.8657     7.8043     3.3855     5.5246 
====================================================================================================

Epoch: 021 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0000     0.0000     0.0000     0.9995     2.6884     2.8657     7.8043     3.3855     5.5246 
====================================================================================================

Epoch: 022 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0000     0.0000     0.0000     0.9995     2.6884     2.8657     7.8043     3.3855     5.5246 
====================================================================================================

Epoch: 023 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0000     0.0000     0.0000     0.9995     2.6884     2.8657     7.8043     3.3855     5.5246 
====================================================================================================

Epoch: 024 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0000     0.0000     0.0000     0.9995     2.6884     2.8657     7.8043     3.3855     5.5246 
====================================================================================================

Epoch: 025 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.0000     0.0000     0.0000     0.9995     2.6884     2.8657     7.8043     3.3855     5.5246 
====================================================================================================

**Comment by vinvino02** [comment]
I have rarely observed these phenomenon in my experiments. Does this happen every time you train? Could you try to start over your training process again?

**Comment by JHCheng111** [comment]
> I have rarely observed these phenomenon in my experiments. Does this happen every time you train? Could you try to start over your training process again?

Thanks for your reply.
I think I have know what caused it. This is where the training went wrong:
Epoch: 009 - 025
====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog 
    0.9085     0.9870     0.9971     0.1018     0.0523     0.3511     0.1265     0.0429     0.1149 
====================================================================================================
Epoch: 010 - 025
 [=============================>]  St 6057/6057  Tot: 41m50s | Rem: 10h28s | Depth Loss: 0.1365 (0.0447)9)
 [=========== 654/654 =========>]  Step: 052ms | Tot: 35s649ms | Rem: 08m34s

====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog
    0.6693     0.9548     0.9930     0.2220     0.1427     0.5392     0.2194     0.0839     0.1783
====================================================================================================


Epoch: 011 - 025
 [============================ 6057/6057 404ms | Tot: 40m47s | Rem: 9h31m | Depth Loss: nan (nan)n))n)31)2)
WARNING:root:NaN or Inf found in input tensor.
 [=========== 654/654 =========>]  Step: 051ms | Tot: 35s674ms | Rem: 07m55s
WARNING:root:NaN or Inf found in input tensor.

====================================================================================================
        d1         d2         d3    abs_rel     sq_rel       rmse   rmse_log      log10      silog
    0.0000     0.0000     0.0000     0.9995     2.6884     2.8657     7.8043     3.3855     5.5246
====================================================================================================
So is that the cause of the loss function or LR schedule? I just found this. Let me do some more research.

**Comment by JHCheng111** [comment]
I think I've figured out why:LR schedule in the code is defined in relation to batch size:
for param_group in optimizer.param_groups:
            if global_step < 2019 * half_epoch:
                current_lr = (1e-4 - 3e-5) * (global_step /
                                              2019/half_epoch) ** 0.9 + 3e-5
2019≈24231/12，so when batch size changes, the constant should changes with it.

**Comment by puyiwen** [comment]
hi,I want to know what is 24231?

**Comment by OriLifschitz** [comment]
Hello, what is 24231/12 please?
is 12 the expected batch size?
What is 24231 then please?

Thanks in advance,
O.

**Comment by puyiwen** [comment]
24231 means the length of train_dataset
&nbsp;


浦一雯
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年1月4日(星期三) 下午5:39
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [vinvino02/GLPDepth] About train process and validate result (Issue #13)





 
Hello, what is 24231/12 please?
 is 12 the expected batch size?
 What is 24231 then please?
 
Thanks in advance,
 O.
